### PR TITLE
feat: Add Google Gemini API plugin

### PR DESCRIPTION
**Pull request by medchedli**
# Motivation

The primary motivation behind this addition is to enhance the content generation capabilities of our application by leveraging Google Generative AI.

## Summary of Changes:
- New Plugin Addition: Introduced a new GeminiPlugin class that extends the BaseBlockPlugin.
- Dependencies: Added @google/generative-ai for leveraging Google Generative AI JS SDK capabilities.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "create-migration": "ts-node ./migrations/config/create.ts"
   },
   "dependencies": {
+    "@google/generative-ai": "^0.19.0",
```
Comment (marrouchi) [inline api/package.json:37]: Thank you @medchedli for your contribution 🥳 

Let's rather create a package.json under src/extensions/plugins/gemini/

In which we would put the dependency. Check the way OpenAI Chatgpt plugin is done 🙏

**Review by marrouchi — CHANGES_REQUESTED** [review]
Can you please add a README file on how to get the token from gemini and explain a bit the settings ?

**Comment by medchedli** [comment]
> Can you please add a README file on how to get the token from gemini and explain a bit the settings ?

Sure, that would be a great addition 👍 I'll maintain this PR as a draft for now and keep improving the implementation.

**Comment by marrouchi** [comment]
Gemini plugin has been pushed in a seperate repo : https://github.com/hexastack/hexabot-plugin-gemini

Just need to do `npx hexabot install plugin Hexastack/hexabot-plugin-gemini`

Will close this one.

**Pull request closed without merging**